### ISSUE TITLE
Using cmake extension modules alongside other extension modules

### ISSUE THREAD

**Issue by Xiretza** [diegoferigo/cmake-build-extension#33]
Unless I'm misinterpreting something, using `BuildExtension` as `cmdclass['build_ext']` in `setup.py` makes any non-cmake extension modules completely ineffective, because `BuildExtension.run()` simply ignores them: https://github.com/diegoferigo/cmake-build-extension/blob/2c3d822e40fe3fd2d769ad09dba367e6617dcc82/src/cmake_build_extension/build_extension.py#L82

Is this correct? How would I use this project if I also have e.g. `cython` extension modules?

**Comment by diegoferigo** [comment]
Hi @Xiretza, thanks for raising this point. I'm not really sure to be honest, I'm not expert in setuptools hooks and I naively assumed that each build extension should process only its own stuff. I imagine that also other build extensions implement their own `BuildExtension.run` method, and I assume that they also filter their instance but I might be wrong.

What's the behavior you're experiencing? Are you sure that the line you linked prevents the execution of other build extensions? Can you check on your example by adding some print statements in all `run` methods of the extensions you're using?

**Comment by Xiretza** [comment]
This project overrides the global `build_ext`:

https://github.com/diegoferigo/cmake-build-extension/blob/2c3d822e40fe3fd2d769ad09dba367e6617dcc82/example/setup.py#L106

As far as I can tell there can only be one `build_ext`, and if it filters out all modules it isn't specifically designed for, that's just not gonna work.

**Comment by Xiretza** [comment]
Not really. `BuildExtension.run()` should probably call `super().run()` at some point, but that will of course try to build and install the `CMakeExtension`s as well.

**Comment by kdewald** [comment]
@diegoferigo you could in theory skip most of the preprocessing that is currently being done in the `run` function and check instance types inside `build_extension` and forwarding to the correct `build_ext` object.

**Comment by yishai1999** [comment]
Any news on this? I just encountered this problem
